### DISCUSSION TITLE
X509SubjectName data to attributes / Xpath importer

### DISCUSSION THREAD

**Discussion by maurotamm** [keycloak/keycloak#36083] [Q&A]
A typical data sent by AD connected SAML clients - they do not send anything in attributes.
Can keycloak extract values from it and map to user attributes?

For example: 
`<NameID Format="urn:oasis:names:tc:SAML:1.1:nameid-format:X509SubjectName">C=DK, O=30808460, CN=TEST MOCES1, Serial=CVR:30808460-RID:53051752</NameID>`

Xpath importer with `//NameID/text()` does nothing. No attribute is created for the user - it likely failed to extract.

I also tried to ensure its not an namespace conflict:
`//*[local-name()='NameID' and namespace-uri()='urn:oasis:names:tc:SAML:2.0:assertion']`
And still nothing.

Does it only support attributes?

I need the X509SubjectName to be available in the tokens later.

**Comment by oising** [comment]
Also came here to ask. There is absolutely no documentation nor examples showing how this is meant to work. Just meaningless tooltips without any context on the mapper inputs. I even trawled the PRs that added this feature, and learned absolutely nothing useful. Incredible.
